internal/httpserve/handlers/webhook.go

```diff
@@ -0,0 +1,89 @@
+package handlers
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/rs/zerolog/log"
+	"github.com/stripe/stripe-go/v81/webhook"
+	echo "github.com/theopenlane/echox"
+
+	ent "github.com/theopenlane/core/internal/ent/generated"
+)
+
+// WebhookReceiverHandler handles incoming stripe webhook events
+func (h *Handler) WebhookReceiverHandler(ctx echo.Context) error {
+	req := ctx.Request()
+	res := ctx.Response()
+
+	const MaxBodyBytes = int64(65536)
+	req.Body = http.MaxBytesReader(res.Writer, req.Body, MaxBodyBytes)
+
+	payload, err := io.ReadAll(req.Body)
+	if err != nil {
+		return ctx.String(http.StatusServiceUnavailable, fmt.Errorf("problem with request. Error: %w", err).Error())
+	}
+
+	event, err := webhook.ConstructEvent(payload, req.Header.Get("Stripe-Signature"), h.Entitlements.Config.StripeWebhookSecret)
+	if err != nil {
+		return ctx.String(http.StatusBadRequest, fmt.Errorf("error verifying webhook signature. Error: %w", err).Error())
+	}
+
+	exists, err := h.checkForEventID(ctx.Request().Context(), event.ID)
```

Comment:
but: you already define `ctx.Request()` as `req` so we should use that
```suggestion
	exists, err := h.checkForEventID(req.Context(), event.ID)
```